Controller/CRUDController.php

```diff
@@ -1129,13 +1121,25 @@ protected function redirectTo($object)
         }
 
         if (!$url) {
-            $url = $this->admin->generateUrl('list');
+            return $this->redirectToList();
         }
 
         return new RedirectResponse($url);
     }
 
     /**
+     * Redirect the user depend on the list view.
+     *
+     * @return RedirectResponse
+     */
+    final protected function redirectToList()
+    {
+        return new RedirectResponse($this->admin->generateUrl('list', [
+            'filter' => $this->admin->getFilterParameters(),
```

Comment:
What about not passing `filter` if the parameters are empty?

Ref: https://github.com/sonata-project/SonataAdminBundle/pull/4741/files#r148931077

Reply:
Fixed that